src/csharp/Grpc.Examples/MathGrpc.cs

```diff
@@ -27,10 +27,35 @@ public static partial class Math
   {
     static readonly string __ServiceName = "math.Math";
 
-    static readonly grpc::Marshaller<global::Math.DivArgs> __Marshaller_math_DivArgs = grpc::Marshallers.Create((arg) => global::Google.Protobuf.MessageExtensions.ToByteArray(arg), global::Math.DivArgs.Parser.ParseFrom);
-    static readonly grpc::Marshaller<global::Math.DivReply> __Marshaller_math_DivReply = grpc::Marshallers.Create((arg) => global::Google.Protobuf.MessageExtensions.ToByteArray(arg), global::Math.DivReply.Parser.ParseFrom);
-    static readonly grpc::Marshaller<global::Math.FibArgs> __Marshaller_math_FibArgs = grpc::Marshallers.Create((arg) => global::Google.Protobuf.MessageExtensions.ToByteArray(arg), global::Math.FibArgs.Parser.ParseFrom);
-    static readonly grpc::Marshaller<global::Math.Num> __Marshaller_math_Num = grpc::Marshallers.Create((arg) => global::Google.Protobuf.MessageExtensions.ToByteArray(arg), global::Math.Num.Parser.ParseFrom);
+    static void __Helper_WriteBufferMessage(global::Google.Protobuf.IMessage message, global::Grpc.Core.SerializationContext context)
+    {
+      #if !GOOGLE_PROTOBUF_DISABLE_BUFFER_SERIALIZATION
+      var bufferMessage = message as global::Google.Protobuf.IBufferMessage;
+      if (bufferMessage != null)
+      {
+        var writer = new global::Google.Protobuf.CodedOutputWriter(context.GetBufferWriter());
+        bufferMessage.WriteTo(ref writer);
+        writer.Flush();
+        context.Complete();
+        return;
+      }
+      #endif
+      context.Complete(global::Google.Protobuf.MessageExtensions.ToByteArray(message));
+    }
+
+    static T __Helper_ParseBufferMessage<T>(global::Grpc.Core.DeserializationContext context, global::Google.Protobuf.MessageParser<T> parser) where T : global::Google.Protobuf.IMessage<T>
```

Comment:
the `T` generic param:
what if I have a protobuf message with field named "T" or a message named "T"? Will that cause name conflict?

Reply:
No conflict. T on the method takes priority in the scope of the method.

https://dotnetfiddle.net/92mrKO